spec/lib/local_authorities_importer_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'rails_helper'
+
+describe 'import of local authorities from MapIt' do
+  let(:source_mapit_data) { File.read(fixture_file('mapit.json')) }
+  let(:mapit_data) { JSON.parse(source_mapit_data) }
+  let(:mapit_url) { "#{MAPIT_BASE_URL}areas/COI,CTY,DIS,LBO,LGD,MTD,UTA" }
+
+  def fixture_file(file)
+    File.expand_path("fixtures/" + file, File.dirname(__FILE__))
+  end
+
+  def stub_mapit_request(data)
+    stub_request(:any, mapit_url).
+      to_return(body: data,
+      status: 200,
+      headers: { 'Content-Length' => data.length })
+  end
+
+  before(:each) do
+    stub_mapit_request(source_mapit_data)
+    LocalAuthoritiesImporter.import_from_mapit
+  end
+
+  it 'imports MapIt formatted json' do
+    expect(LocalAuthority.count).to eq(8)
+
+    la = LocalAuthority.find_by(gss: 'S12000033')
+
+    expect(la.name).to eq("Aberdeen City Council")
+    expect(la.snac).to eq("00QA")
+    expect(la.tier).to eq("unitary")
+  end
+
+  it 'updates name and tier fields only' do
+    updated_name_type_and_ons = '{
+      "9999": {
+        "parent_area": null,
+        "generation_high": 1,
+        "all_names": {},
+        "id": 9999,
+        "codes": {
+            "ons": "XXXX",
+            "gss": "S12000033",
+            "unit_id": "30421"
+        },
+        "name": "A Different Council",
+        "country": "S",
+        "type_name": "Unitary Authority",
+        "generation_low": 1,
+        "country_name": "Scotland",
+        "type": "DIS"
+      }
+    }'
+
+    stub_mapit_request(updated_name_type_and_ons)
+
+    LocalAuthoritiesImporter.import_from_mapit
+
+    expect(LocalAuthority.count).to eq(8)
+
+    la = LocalAuthority.find_by(gss: 'S12000033')
+
+    expect(la.name).to eq("A Different Council")
+    expect(la.snac).to_not eq("XXXX")
```

Comment:
Why don't we want to update the SNAC? They're unlikely to change in practice since they aren't being issued any more, but we might need to correct a SNAC in mapit or add one which is missing for some reason, and I think we'd want that to propagate to this app, unless I've missed something.